pkg/receive/handler.go

```diff
@@ -403,6 +405,13 @@ func (h *Handler) handleRequest(ctx context.Context, rep uint64, tenant string,
 	return h.forward(ctx, tenant, r, wreq)
 }
 
+func (h *Handler) isTenantValid(tenant string, err error, w http.ResponseWriter) {
+	if tenant != path.Base(tenant) {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
Should be server error? 🤔 Or bad request?

Reply:
hmm that's a good point. It could be either but in any case I'll go with Saswata's suggestion, stay tuned!